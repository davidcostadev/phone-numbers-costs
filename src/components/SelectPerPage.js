import React from 'react';
import PropTypes from 'prop-types';
import Select from './forms/Select';
import { paginatorType } from '../types';

const options = [10, 50, 100, 250, 500].map(option => ({
  value: option,
  text: option,
}));

const SelectPerPage = ({ onChangePerPage, paginator }) => {
  if (!paginator) {
    return null;
  }

  return (
    <div className="select-per-page">
      <Select
        id="per-page"
        options={options}
        value={paginator.perPage}
        onChange={({ target }) => onChangePerPage(target.value)}
      />
    </div>
  );
};

SelectPerPage.propTypes = {
  onChangePerPage: PropTypes.func.isRequired,
  paginator: paginatorType,
};

SelectPerPage.defaultProps = {
  paginator: null,
};


export default SelectPerPage;
