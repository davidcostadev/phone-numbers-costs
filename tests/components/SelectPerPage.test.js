import React from 'react';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SelectPerPage from '../../src/components/SelectPerPage';

configure({ adapter: new Adapter() });

describe('SelectPerPage Component', () => {
  let onChangePerPage;

  beforeEach(() => {
    onChangePerPage = jest.fn();
  });

  describe('Render', () => {
    it('default', () => {
      const tree = renderer.create((
        <SelectPerPage onChangePerPage={onChangePerPage} />
      )).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('with props', () => {
      const tree = renderer.create((
        <SelectPerPage value={50} onChangePerPage={onChangePerPage} />
      )).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Behavior', () => {
    it('click on onChangePerPage', () => {
      const wrapper = shallow(<SelectPerPage onChangePerPage={onChangePerPage} />);

      wrapper.find('#per-page').simulate('change', { target: { value: 50 } });
      expect(onChangePerPage).toBeCalledWith(50);
    });
  });
});
