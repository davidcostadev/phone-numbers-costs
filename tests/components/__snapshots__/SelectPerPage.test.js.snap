// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectPerPage Component Render default 1`] = `null`;

exports[`SelectPerPage Component Render with props 1`] = `
<div
  className="select-per-page"
>
  <select
    id="per-page"
    onChange={[Function]}
    value={50}
  >
    <option
      value={10}
    >
      10
    </option>
    <option
      value={50}
    >
      50
    </option>
    <option
      value={100}
    >
      100
    </option>
    <option
      value={250}
    >
      250
    </option>
    <option
      value={500}
    >
      500
    </option>
  </select>
</div>
`;
